src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipePayloadDecoder.cs

```diff
@@ -113,26 +113,26 @@ internal static object[] DecodePayload(ref EventSource.EventMetadata metadata, R
                     // Try to find null terminator (0x00) from the byte span
                     // NOTE: we do this by hand instead of using payload.IndexOf('\n') because payload may be unaligned due to 
                     // mixture of different types being stored in the same buffer. (see eventpipe.cpp:CopyData)
-                    int byteCount = -1;
-                    for (int j = 1; j < payload.Length; j+=2)
+                    int nullByteIdx = -1;
+                    for (int j = 0; j < payload.Length; j+=2)
                     {
-                        if (payload[j-1] == (byte)(0) && payload[j] == (byte)(0))
+                        if (payload[j] == (byte)(0) && payload[j+1] == (byte)(0))
```

Comment:
Unfortunately now the code will read off the end of the buffer when given an odd size buffer (which isn't supposed to happen, but depending on global correctness of code to prevent overruning a buffer isn't great)

I'd suggest leaving the code how it was before, except set byteCount = j+1 and adjust the payload slice length computations.